android/src/main/kotlin/studio/midoridesign/gal/GalPlugin.kt

```diff
@@ -0,0 +1,304 @@
+package studio.midoridesign.gal
+
+import android.Manifest
+import android.app.Activity
+import android.content.ContentResolver
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.os.Environment
+import android.os.Handler
+import android.os.Looper
+import android.provider.MediaStore
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.PluginRegistry
+import org.apache.commons.imaging.ImageFormat
+import org.apache.commons.imaging.Imaging
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.util.Locale
+
+class GalPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+    PluginRegistry.RequestPermissionsResultListener {
+    private var channel: MethodChannel? = null
+    private var pluginBinding: FlutterPlugin.FlutterPluginBinding? = null
+    private var activity: Activity? = null
+    private var requestAccessCallback: Runnable? = null
+    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "gal")
+        channel?.setMethodCallHandler(this)
+        pluginBinding = flutterPluginBinding
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel?.setMethodCallHandler(null)
+        pluginBinding = null
+    }
+
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "putVideo", "putImage" -> {
+                Thread {
+                    try {
+                        val path = call.argument<String>("path")
+                        if (path == null) {
+                            result.error(
+                                "PATH_IS_REQUIRED",
+                                "The path argument is required",
+                                null,
+                            )
+                            return@Thread
+                        }
```

Comment:
This technically can be null so it's better to handle it, it will not because the original caller in Dart doesn't allow it to be null

Reply:
> This technically can be null so it's better to handle it, it will not because the original caller in Dart doesn't allow it to be null

`path` is required in the Dart calling function. Are you saying you want to handle empty?

Reply:
The plugin code doesn't know the path can't be null from the dart code so it can be null even though it's impossible to reach this in our case